src/ArrayManifold.jl

```diff
@@ -12,6 +12,10 @@ end
 convert(::Type{M},m::ArrayManifold{M}) where M <: Manifold = m.manifold
 convert(::Type{ArrayManifold{M}},m::M) where M <: Manifold = ArrayManifold(M)
 
+function representation_size(M::ArrayManifold, ::Type{T}) where {T}
```

Comment:
Neither this nor the `manifold_dimension` dimension below should be necessary. Just implement the generic `IsDecoratorManifold` version in `Manifolds.jl`:

```julia
@traitfn function representation_size(M::MT, ::Type{T}) where {MT<:Manifold,T;!IsDecoratorManifold{MT}}
    error("representation_size not implemented for manifold $(typeof(M)) and type $(T).")
end

@traitfn function representation_size(M::MT, ::Type{T}) where {MT<:Manifold,T;IsDecoratorManifold{MT}}
    representation_size(base_manifold(M), T)
end
```

Reply:
Thanks, good idea.